crates/proof-of-sql/src/sql/postprocessing/evaluate.rs

```diff
@@ -0,0 +1,267 @@
+use super::{PostprocessingError, PostprocessingResult};
+use crate::{
+    base::{
+        database::{ColumnType, OwnedColumn, OwnedTable},
+        math::decimal::{scale_scalar, try_into_to_scalar, Precision},
+        scalar::Scalar,
+    },
+    sql::ast::{
+        add_subtract_owned_columns, multiply_owned_columns, try_add_subtract_column_types,
+        try_divide_column_types, try_divide_owned_columns, try_multiply_column_types,
+    },
+};
+use core::cmp::Ordering;
+use proof_of_sql_parser::{
+    intermediate_ast::{BinaryOperator, Expression, Literal, UnaryOperator},
+    Identifier,
+};
+
+/// Evaluator that evaluates a `proof_of_sql_parser::intermediate_ast::Expression`
+/// on an `OwnedTable` and returns an `OwnedColumn`.
+pub struct PostprocessingEvaluator<S: Scalar> {
+    table: OwnedTable<S>,
+    _in_agg_scope: bool,
+}
+
+impl<S: Scalar> PostprocessingEvaluator<S> {
+    /// Creates a new `PostprocessingEvaluator` with the given column mapping.
+    pub fn new(table: OwnedTable<S>) -> Self {
+        Self {
+            table,
+            _in_agg_scope: false,
+        }
+    }
+    /// Creates a new `PostprocessingEvaluator` with the given column mapping and within aggregation scope.
+    #[allow(dead_code)]
+    pub(crate) fn new_agg(table: OwnedTable<S>) -> Self {
+        Self {
+            table,
+            _in_agg_scope: true,
+        }
+    }
```

Comment:
is there a good reason to allow unused functions and fields here? If they will end up being used in a future PR then it seems fine

Reply:
Yes. I will add aggregation once `SelectExpr` is added in the next PR